mlir/Makefile

```diff
@@ -28,6 +28,20 @@ help:
 .PHONY: all
 all: llvm mhlo dialects enzyme
 
+.PHONY: patch
```

Comment:
Could you please add the help (or doc?) string explaining when to call this rule?

Reply:
I'll wait until the discussion for this branch is finished to update the docs, but thanks for pointing it out!